internal/dag/dag.go

```diff
@@ -30,6 +30,11 @@ import (
 // between Kubernetes Ingress objects, the backend Services, and Secret objects.
 // The DAG models these relationships as Roots and Vertices.
 type DAG struct {
+	// IngressRouteRootNamespaces specifies the namespaces where root
+	// IngressRoutes can be defined. If empty, roots can be defined in any
+	// namespace.
+	IngressRouteRootNamespaces []string
```

Comment:
I am wondering if there's a better place for this, as this is adding business logic that doesn't have much to do with building a DAG.

Reply:
> Adding and having to remove later
Not sure I am following.

I had thought about building a ResourceEventHandler that would wrap the DAG and implement this logic. Is that what you have in mind?

Reply:
What I meant is you could load up the DAG with all the items, then have another process validate that those items should exist in there. I think it's better to remove up front rather than post-processing. 

The same issues will come up later with validating duplicate healthchecks or strategy, etc. Do we look for those while creating the DAG or somehow after.

Reply:
OK, I think I am with you, and I think that is what I am doing in this case, right? When we are recomputing the DAG, we are checking whether the IngressRoute can exist in the given namespace. In the case that it cannot, we skip processing it and thus do not add it to the DAG.

Reply:
yup 100% you are doing that! Was just replying to your comment about "Is this the right place for it".

Reply:
Hehe cool! 

I still think we might want to put something between the watchers and the DAG, and move the validation logic there. Not sure if it's worth it though.

Reply:
This probably should live in the same place that we do a check for ingress class .. but we don't do that at all yet.

I'd try pushing that logic into `contour.DAGAdapter`.